HKMP/Game/Client/PlayerManager.cs

```diff
@@ -175,29 +301,101 @@ internal class PlayerManager {
             _skinManager.ResetLocalPlayerSkin();
         }
 
-        // TODO: investigate whether it is better to disable/setActive(false) player objects instead of destroying
         // and only destroy when player left server
         /// <summary>
         /// Destroy the player with the given ID.
         /// </summary>
         /// <param name="id">The player ID.</param>
-        public void DestroyPlayer(ushort id) {
+        public void RecyclePlayer(ushort id) {
             if (!_playerData.TryGetValue(id, out var playerData)) {
-                Logger.Get().Warn(this, $"Tried to destroy player that does not exists for ID {id}");
+                Logger.Get().Warn(this, $"Tried to recycle player that does not exists for ID {id}");
                 return;
             }
 
-            // Destroy gameObject and remove from mapping
-            Object.Destroy(playerData.PlayerContainer);
+            // Recycle gameObject
+            playerData.PlayerContainer = null;
+            _activePlayers.TryGetValue(id, out var container);
+            _activePlayers.Remove(id);
+            container?.SetActive(false);
+            _inactivePlayers.Add(container);
         }
 
         /// <summary>
-        /// Destroy all existing players.
+        /// Recycle all existing players.
         /// </summary>
-        public void DestroyAllPlayers() {
-            foreach (var playerData in _playerData.Values) {
-                // Destroy gameObject
-                Object.Destroy(playerData.PlayerContainer);
+        public void RecycleAllPlayers() {
+            foreach (var (id, playerData) in _playerData) {
```

Comment:
You are not using one of the variables from the deconstruction.
```suggestion
            foreach (var (id, _) in _playerData) {
```